### DISCUSSION TITLE
Help in Installation - ImportError: cannot import name 'base_impl

### DISCUSSION THREAD

**Discussion by robodevchampion** [oracle/python-oracledb#248] [Q&A]
Hi, 

I wanted to use the library for an project, for which development I am doing on VDI Machine. Due to some firewall issue, I can not directly install library using `pip install `command, what I do is basically I download library from git, unzip it and use it. 

Some how when I download this library from git and unzip it and tried to use it, I get below error

**ImportError: cannot import name 'base_impl' from partially initialized module 'oracledb'**

If you can help me in fixing this, that will be really helpful.

Thanks

**Comment by anthony-tuininga** [comment]
I would suggest downloading a wheel directly from [PyPI](https://pypi.org/project/oracledb/#files). Choose the right Python version for your platform and then install directly from the downloaded file (``python -m pip install <file_name>``). If you need to install from a GitHub downloaded zip, follow the [documentation](https://python-oracledb.readthedocs.io/en/latest/user_guide/installation.html#installing-from-source-code).

If none of those suggestions help, please provide more detail about your configuration/platform and what steps you took.

**Reply by kishorkumarg** [reply]
Hello @abhibhatia98, is this issue resolved, can you please help me, I am facing the same issue, trying to fix since two days, but could not able to. 
please help me to fix this...!
Thanks in advance
